codecs/avif_enc/avif_enc.cpp

```diff
@@ -0,0 +1,88 @@
+#include "avif/avif.h"
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+using namespace emscripten;
+
+struct AvifOptions {
+  // [0 - 63]
+  // 0 = lossless
+  // 63 = worst quality
+  int minQuantizer;
+  int maxQuantizer;
+  // [0 - 6]
+  // Creates 2^n tiles in that dimension
+  int tileRowsLog2;
+  int tileColsLog2;
+  // [0 - 10]
+  // 0 = slowest
+  // 10 = fastest
+  int speed;
+  // 0 = 4:2:0
+  // 1 = 4:2:2
+  // 2 = 4:4:4
+  int subsample;
+};
+
+avifRWData output = AVIF_DATA_EMPTY;
+val encode(std::string buffer, int width, int height, AvifOptions options) {
+  int depth = 8;
+  avifPixelFormat format;
+  switch (options.subsample) {
+  case 0:
+    format = AVIF_PIXEL_FORMAT_YUV420;
+    break;
+  case 1:
+    format = AVIF_PIXEL_FORMAT_YUV422;
+    break;
+  case 2:
+    format = AVIF_PIXEL_FORMAT_YUV444;
+    break;
+  }
+  avifImage *image = avifImageCreate(width, height, depth, format);
+
+  uint8_t *rgba = (uint8_t *)buffer.c_str();
+
+  avifImageAllocatePlanes(image, AVIF_PLANES_RGB);
+  avifImageAllocatePlanes(image, AVIF_PLANES_A);
+
+  for (int y = 0; y < height; y++) {
```

Comment:
As far as I can tell, this is not necessary and we can assign `pixels` directly to our data without allocating an extra copy and copying everything byte-by-byte. https://github.com/AOMediaCodec/libavif/blob/master/include/avif/avif.h#L371-L410